File: python/rna-transcription/rna_transcription.py
"""
rna_transcription - Given a DNA strand, return its RNA complement
                    (per RNA transcription).
"""

def to_rna(dna):
    complement = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'
    }

    # validate input (only contains GCTA)
    valid = set('GCTA')

    if set(dna) <= valid:
        return ''.join([complement[nuc] for nuc in dna])
    else:
        return ''

File: javascript/rna-transcription/rna-transcription.js
/*

rna-transcription - given a dna strand, return its RNA complement
                    (per RNA transcripton)

*/

var DnaTranscriber = function() {}

DnaTranscriber.prototype.toRna = function(dna) {
  // validate input, make sure it only matches GCTA characters
  if (!dna.match(/^[GCTA]+$/)) {
    throw new Error("Invalid input");
  }

  // dna to rna transcription object
  var transcription = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  };

  var rna = '';

  // loop over dna nucleotides and transcribe into rna
  for (var i = 0; i < dna.length; i++) {
    rna += transcription[dna.charAt(i)];
  }

  return rna;
}

module.exports = DnaTranscriber;
